paymentsheet/src/main/java/com/stripe/android/common/analytics/experiment/LogLinkGlobalHoldbackExposure.kt

```diff
@@ -1,35 +1,112 @@
+import com.stripe.android.paymentsheet.injection.LinkDisabledApiRepository
+import com.stripe.android.paymentsheet.repositories.CustomerRepository
+import com.stripe.android.paymentsheet.state.PaymentElementLoader
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
 
-internal class LogLinkGlobalHoldbackExposure @Inject constructor(
+internal interface LogLinkGlobalHoldbackExposure {
+    /**
+     * Logs the exposure of the Link Global Holdback experiment.
+     *
+     * @param elementsSession The session containing the experiment data.
+     * @param state The current state of the payment element loader.
+     */
+    operator fun invoke(
+        elementsSession: ElementsSession,
+        state: PaymentElementLoader.State
+    )
+}
+
+internal class DefaultLogLinkGlobalHoldbackExposure @Inject constructor(
     private val eventReporter: EventReporter,
+    @LinkDisabledApiRepository private val linkDisabledApiRepository: LinkRepository,
+    @IOContext private val workContext: CoroutineContext,
+    private val customerRepository: CustomerRepository,
     private val logger: Logger
-) {
+) : LogLinkGlobalHoldbackExposure {
 
-    operator fun invoke(
+    override operator fun invoke(
         elementsSession: ElementsSession,
-    ) = runCatching {
+        state: PaymentElementLoader.State
+    ) {
         if (FeatureFlags.linkGlobalHoldbackExposureEnabled.isEnabled) {
-            val experimentsData = requireNotNull(
-                elementsSession.experimentsData
-            ) { "Experiments data required to log exposures" }
-            val holdbackOn = elementsSession.linkSettings?.linkGlobalHoldbackOn == true
-            eventReporter.onExperimentExposure(
-                experiment = LinkGlobalHoldback(
-                    arbId = experimentsData.arbId,
-                    group = if (holdbackOn) TREATMENT else CONTROL,
-                ),
-            )
+            CoroutineScope(workContext).launch {
+                runCatching {
+                    logExposure(elementsSession, state)
+                }.onFailure { error ->
+                    logger.error("Failed to log Global holdback exposure", error)
+                }
+            }
         }
-    }.onFailure { error ->
-        logger.error("Failed to log Global holdback exposure", error)
     }
+
+    private suspend fun logExposure(
+        elementsSession: ElementsSession,
+        state: PaymentElementLoader.State
+    ) {
+        val experimentsData = requireNotNull(
+            elementsSession.experimentsData
+        ) { "Experiments data required to log exposures" }
+        val experimentGroup: String = elementsSession.experimentsData
+            ?.experimentAssignments[LINK_GLOBAL_HOLD_BACK] ?: "control"
+
+        val customerEmail = state.getEmail()
+
+        val isReturningUser: Boolean = customerEmail?.let { isReturningUser(email = it) } == true
```

Comment:
Nit: We can use to compiler to our advantage and clean this up a bit.

```suggestion
        val isReturningUser = customerEmail != null && isReturningUser(customerEmail)
```